standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RetryingHMSHandler.java

```diff
@@ -227,6 +237,23 @@ public Result invokeInternal(final Object proxy, final Method method, final Obje
     }
   }
 
+  private boolean isRecoverableException(Throwable t) {
```

Comment:
The RPC flow in HMS looks like:
**RetryingMetaStoreClient <---> RetryingHMSHandler <---> DBMS**

Such recoverable exceptions always occur between `RetryingHMSHandler` and `DBMS`, I think it's RetryingHMSHandler's duty to handle this retry strategy. And this can also reduce the reconnection from metastore client to metastore server.

Reply:
👍